metadata/src/main/java/org/apache/kafka/image/AclsDelta.java

```diff
@@ -61,7 +61,13 @@ public void replay(AccessControlEntryRecord record) {
     }
 
     public void replay(RemoveAccessControlEntryRecord record) {
-        changes.put(record.id(), Optional.empty());
+        if (image.acls().containsKey(record.id())) {
+            changes.put(record.id(), Optional.empty());
+        } else if (changes.containsKey(record.id())) {
+            changes.remove(record.id());
```

Comment:
Btw, the Jira mentions that we don't have "remove acl" to be idempotent. "Remove acl" is idempotent if they are process in the same batch/delta.

Reply:
You are correct. I missed the `else if`.